src/audio-worklet/basic/audio-worklet-node-options/main.js

```diff
@@ -3,30 +3,39 @@
 // found in the LICENSE file.
 
 const audioContext = new AudioContext();
+let oscillatorProcessor;
 
 const startAudio = async (context, options) => {
-  await context.audioWorklet.addModule('oscillator-processor.js');
-  const oscillatorProcessor =
+  if (!oscillatorProcessor) {
+    await context.audioWorklet.addModule('oscillator-processor.js');
+    oscillatorProcessor =
       new AudioWorkletNode(context, 'oscillator-processor', {
         processorOptions: {
           waveformType: options.waveformType,
           frequency: options.frequency,
-        }});
-  oscillatorProcessor.connect(context.destination);
+        }
+      });
+    oscillatorProcessor.connect(context.destination);
+  }
 };
 
-// A simplem onLoad handler. It also handles user gesture to unlock the audio
+// A simple onLoad handler. It also handles user gesture to unlock the audio
 // playback.
 window.addEventListener('load', async () => {
   const buttonEl = document.getElementById('button-start');
   buttonEl.disabled = false;
+
   buttonEl.addEventListener('click', async () => {
-    const waveformType =
-        document.querySelector('#demo-select-waveform-type').value;
-    const frequency = document.querySelector('#demo-input-frequency').value;
-    await startAudio(audioContext, {waveformType, frequency});
-    audioContext.resume();
-    buttonEl.disabled = true;
-    buttonEl.textContent = 'Playing...';
+    if (!oscillatorProcessor) { // If audio is not playing, start audio
```

Comment:
nit: Let's move the comment below the `if()`.

Reply:
Done.